src/common/types/cpuArchitecture.ts

```diff
@@ -46,23 +46,41 @@ export const getAllCpuArchitectures = (): SupportedCpuArchitecture[] => [
 export const getSupportedCpuArchitectures = (
   canSelectCpuArch: boolean,
   cpuArchitectures: CpuArchitecture[],
+  day1CpuArchitecture?: SupportedCpuArchitecture,
 ): SupportedCpuArchitecture[] => {
   const newSupportedCpuArchs: SupportedCpuArchitecture[] = [];
-  if (cpuArchitectures) {
-    cpuArchitectures.forEach((cpuArch) => {
-      if (
-        (cpuArch === CpuArchitecture.ppc64le || cpuArch === CpuArchitecture.s390x) &&
-        canSelectCpuArch
-      ) {
-        newSupportedCpuArchs.push(cpuArch);
-      } else if (
-        cpuArch !== CpuArchitecture.MULTI &&
-        cpuArch !== CpuArchitecture.USE_DAY1_ARCHITECTURE
-      ) {
-        newSupportedCpuArchs.push(cpuArch);
-      }
-    });
+  if (day1CpuArchitecture) {
+    if (day1CpuArchitecture === CpuArchitecture.ppc64le && canSelectCpuArch) {
+      newSupportedCpuArchs.push(CpuArchitecture.ppc64le);
+    } else if (canSelectCpuArch) {
+      cpuArchitectures.forEach((cpuArch) => {
+        if (
+          cpuArch !== CpuArchitecture.MULTI &&
+          cpuArch !== CpuArchitecture.USE_DAY1_ARCHITECTURE &&
+          cpuArch !== CpuArchitecture.ppc64le
+        ) {
+          newSupportedCpuArchs.push(cpuArch);
+        }
+      });
+    }
+  } else {
+    if (cpuArchitectures) {
```

Comment:
Why are you checking for truthiness? It says on the signature that it is a CpuArchitecture[]

Reply:
I've changed it